Update zha.markdown with note about changing Zigbee channel (#18865)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

### Defining Zigbee channel to use

> ZHA prefers to use Zigbee channel 15 by default. You can change this using YAML configuration, however, note that changing the channel requires repairing all devices.

Reviewer: ```suggestion
ZHA prefers to use Zigbee channel 15 by default. You can change this using YAML configuration, but this only works
if there's no existing network. To change the channel for an existing channel, radio has to be factory reset and a new network to be formed. This requires re-pairing of all the devices.
```

> Zigbee network channel changes 15 will be used by default by ZHA but the Zigbee network channel can be changed via YAML, however note that changing Zigbee network channel changes requires repairing all already devices paired to ZHA.

Reviewer 2: ```suggestion
ZHA prefers to use Zigbee channel 15 by default. You can change this using YAML configuration, however, note that changing the channel requires repairing all devices.
```

Reviewer 3: "repairing all already devices" ? what is this?

Reviewer: Because currently it requires killing the current network and forming a new one. But here's the challenge: re-forming the network is radio specific and now we run into the same issue when this is now not ZHA integration specific, but radio specific and shouldn't really belong to ZHA integration documentation. 

Right now channel configuration is taken into consideration only for new sticks. if ZHA was configured at least once and already formed the network on the stick, then channel configuration is ignored.

Reviewer 2: So, shall we just close this instead?

### Defining Zigbee channel to use

ZHA prefers to use Zigbee channel 15 by default. You can change this using YAML configuration, but this only works

> if there's no existing network. To change the channel for an existing channel, radio has to be factory reset and a new network to be formed. This requires re-pairing of all the devices.

Reviewer: Doh
```suggestion
if there's no existing network. To change the channel for an existing network, radio has to be factory reset and a new network to be formed. This requires re-pairing of all the devices.
```

      channels: [15, 20, 25]  # Channel mask
```

> Tip: Reference this for channel selection [Zigbee and WiFi coexistence (using non-overlapping frequency-bands)](https://support.metageek.com/hc/en-us/articles/203845040-ZigBee-and-WiFi-Coexistence).

Reviewer 2: I don't think this is an improvement, and quite an unneeded change. Please revert it.

Reviewer 2: ```suggestion
This is a good reference for channel selection [Zigbee and Wifi Coexistance](https://support.metageek.com/hc/en-us/articles/203845040-ZigBee-and-WiFi-Coexistence)
```
